### ISSUE TITLE
RGA appears to mistake SPIR-V for GLSL

### ISSUE THREAD

**Issue by jbarczak** [GPUOpen-Tools/RGA#15]
It appears that RGA sometimes mis-interprets SPIR-V as GLSL when given SPIR-V input.   Here is the result of compiling a pair of spir-v files.  Both were generated by the glslang HLSL front-end, and pass spirv-val.  One works as expected.  The other throws a GLSL parse error

[scratch.zip](https://github.com/GPUOpen-Tools/RGA/files/1271839/scratch.zip)

```
c:\scratch>"c:\Program Files\GPUOpen\Radeon GPU Analyzer\rga.exe"  -s vulkan-spv good.spv --isa ".\pyramid.isa"  -a ".\pyramid.analysis"  --il ".\pyramid.il"   -c Baffin

Target GPU detected:

Baffin (Graphics IP v8)
        AMD Radeon (TM) E9260
        AMD Radeon Pro 450
        AMD Radeon Pro 455
        AMD Radeon Pro 460
        AMD Radeon Pro 465
        Radeon (TM) Pro V5300X
        Radeon (TM) Pro WX 4100
        Radeon (TM) Pro WX Series
        Radeon RX 550 Series
        Radeon RX 560 Series
        Radeon(TM) RX 460 Graphics

Building for Baffin... succeeded.

c:\scratch>"c:\Program Files\GPUOpen\Radeon GPU Analyzer\rga.exe"  -s vulkan-spv bad.spv --isa ".\pyramid.isa"  -a ".\pyramid.analysis"  --il ".\pyramid.il"   -c Baffin

Target GPU detected:

Baffin (Graphics IP v8)
        AMD Radeon (TM) E9260
        AMD Radeon Pro 450
        AMD Radeon Pro 455
        AMD Radeon Pro 460
        AMD Radeon Pro 465
        Radeon (TM) Pro V5300X
        Radeon (TM) Pro WX 4100
        Radeon (TM) Pro WX Series
        Radeon RX 550 Series
        Radeon RX 560 Series
        Radeon(TM) RX 460 Graphics

Building for Baffin... failed.
Compiling fragment stage:
ERROR: #version: Desktop shaders for Vulkan SPIR-V require version 140 or higher
ERROR: 0:1: '' :  syntax error, unexpected IDENTIFIER
ERROR: 2 compilation errors.  No code generated.
```

**Comment by AmitBM** [comment]
Hi jbarczak,
The error message that you saw here was incorrect, and was caused by a bug that was fixed in [RGA 1.3](https://github.com/GPUOpen-Tools/RGA/releases/tag/1.3.0.0).
However, we are still having trouble compiling the SPIR-V binary that you provided.
Can you please share more information about the front-end compiler that was used to generate this binary, and perhaps attach the high-level source code?

**Comment by jbarczak** [comment]
I'll follow up if I'm able to reproduce this again.  If I remember correctly I compiled this using whatever version of glslang is currently embedded in my 'pyramid' tool.   https://github.com/jbarczak/Pyramid  If I can reproduce this again I'll provide more info..
